src/stim/simulators/tableau_simulator.pybind.cc

```diff
@@ -1475,13 +1500,25 @@ void stim_pybind::pybind_tableau_simulator_methods(pybind11::module &m, pybind11
 
     c.def(
         "copy",
-        [](const PyTableauSimulator &self) {
+        [](const PyTableauSimulator &self, bool copy_rng) {
+            if (copy_rng) {
+                return PyTableauSimulator(self, std::make_shared<std::mt19937_64>(*self.rng_reference));
+            }
             PyTableauSimulator copy = self;
```

Comment:
I think we need to undo the "shared rng" thing here. This is a holdover from when there was one global rng generator.

Working on the copy should not change the original in any way. Now is the time to fix these semantics since effectively thru are becoming visible for the first time.

Reply:
Agreed & done. I'll do this in other classes (seems there is only one: frame sim) as follow-up.